secrets/kubernetes/ondemand.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/prometheus/prometheus/secrets"
+)
+
+type OnDemandSPConfig struct {
+	ClientConfig
+}
+
+// Name returns the name of the Config.
+func (*OnDemandSPConfig) Name() string { return "kubernetes_on_demand" }
+
+// NewDiscoverer returns a Discoverer for the Config.
+func (c *OnDemandSPConfig) NewProvider(ctx context.Context, opts secrets.ProviderOptions) (secrets.Provider[SecretConfig], error) {
```

Comment:
Do we need on demand one? Seems unused? 🤔

Reply:
That's true, it was mostly a reference implementation. I will remove it!